pallets/migration/src/lib.rs

```diff
@@ -0,0 +1,355 @@
+			accounts: Vec<(Vec<u8>, Vec<u8>)>,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			let num_accounts = accounts.len();
+			ensure!(
+				accounts.len()
+					<= T::MigrationMaxAccounts::get()
+						.try_into()
+						.map_err(|_| ArithmeticError::Overflow)?,
+				Error::<T>::TooManyAccounts
+			);
+
+			for (key, value) in accounts {
+				storage::unhashed::put_raw(key.as_slice(), value.as_slice());
+			}
+
+			// This is safe as MigrationMaxAccounts is a u64
+			Self::deposit_event(Event::<T>::MigratedSystemAccounts(num_accounts as u64));
+
+			Ok(
+				Some(<T as pallet::Config>::WeightInfo::migrate_system_account(
+					num_accounts as u64,
+				))
+				.into(),
+			)
+		}
+
+		/// Migrates a the `TotalIssuance`.
+		///
+		/// The provide balance here, will be ADDED to the existing `TotalIssuance` of the system.
+		/// Calley better be sure, that the total issuance matches the actual total issuance in the system,
+		/// which means, that the `AccountInfo` from the frame_system is migrated afterwards.
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::migrate_balances_issuance())]
+		#[transactional]
+		pub fn migrate_balances_issuance(
+			origin: OriginFor<T>,
+			additional_issuance: T::Balance,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			let current_issuance = pallet_balances::Pallet::<T>::total_issuance();
+			let total_issuance = current_issuance.saturating_add(additional_issuance);
+
+			let key = <pallet_balances::pallet::TotalIssuance<T> as frame_support::storage::generator::StorageValue<T::Balance>>::storage_value_final_key();
+
+			storage::unhashed::put_raw(&key[..], total_issuance.encode().as_slice());
+
+			Self::deposit_event(Event::<T>::MigratedTotalIssuance(
+				current_issuance,
+				total_issuance,
+			));
+
+			Ok(().into())
+		}
+
+		/// Migrates vesting information to this system.
+		///
+		/// The `VestingInfo` is adapted off-chain, so that it represents the correct vesting information
+		/// on this chain.
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::migrate_vesting_vesting(T::MigrationMaxVestings::get()))]
+		#[transactional]
+		pub fn migrate_vesting_vesting(
+			origin: OriginFor<T>,
+			vestings: Vec<(T::AccountId, VestingInfo<BalanceOf<T>, T::BlockNumber>)>,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			ensure!(
+				vestings.len()
+					<= T::MigrationMaxVestings::get()
+						.try_into()
+						.map_err(|_| ArithmeticError::Overflow)?,
+				Error::<T>::TooManyVestings
+			);
+
+			// This is safe as MigrationMaxVestings is a u64
+			let mut trying = vestings.len() as u64;
+			let num_vestings = trying;
+
+			for (who, schedule) in vestings {
+				let _not_care_here = pallet_vesting::Pallet::<T>::add_vesting_schedule(
+					&who,
+					schedule.locked,
+					schedule.per_block,
+					schedule.starting_block,
+				)
+				.map_err(|_| {
+					Self::deposit_event(Event::<T>::FailedToMigrateVestingFor(who.clone()));
```

Comment:
No, need an event here. Or is there another way to deposit errors?

Reply:
I meant sending a error as part of event itself since this is deposited when migration failed. I think DispatchError should work here

Reply:
The idea is to use an event so that one failed entry doesn't blow up the entire transactions. We'll be able to look at the events and manually fix just those vesting configurations using `sudo`.